osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -441,138 +458,123 @@ private void updateState(CatcherAnimationState state)
+                case Fruit _:
+                    return caughtFruitPool.Get();
 
-                case Fruit fruit:
-                    return new DrawableFruit(fruit);
+                case Banana _:
+                    return caughtBananaPool.Get();
 
-                case TinyDroplet tiny:
-                    return new DrawableTinyDroplet(tiny);
-
-                case Droplet droplet:
-                    return new DrawableDroplet(droplet);
+                case Droplet _:
+                    return caughtDropletPool.Get();
 
                 default:
                     return null;
             }
         }
 
-        private void clearPlate(DroppedObjectAnimation animation)
+        private CaughtObject getDroppedObject(CaughtObject caughtObject)
         {
-            var caughtObjects = caughtFruitContainer.Children.ToArray();
-            caughtFruitContainer.Clear(false);
+            var droppedObject = getCaughtObject(caughtObject.HitObject);
 
-            droppedObjectTarget.AddRange(caughtObjects);
+            droppedObject.CopyStateFrom(caughtObject);
+            droppedObject.Anchor = Anchor.TopLeft;
+            droppedObject.Position = caughtFruitContainer.ToSpaceOfOtherDrawable(caughtObject.DrawPosition, droppedObjectTarget);
 
-            foreach (var caughtObject in caughtObjects)
-                drop(caughtObject, animation);
+            return droppedObject;
         }
 
-        private void removeFromPlate(DrawablePalpableCatchHitObject caughtObject, DroppedObjectAnimation animation)
+        private void clearPlate(DroppedObjectAnimation animation)
         {
-            if (!caughtFruitContainer.Remove(caughtObject))
-                throw new InvalidOperationException("Can only drop a caught object on the plate");
+            var caughtObjects = caughtFruitContainer.Children.ToArray();
+            var droppedObjects = caughtObjects.Select(getDroppedObject).ToArray();
+
+            caughtFruitContainer.Clear(false);
 
-            droppedObjectTarget.Add(caughtObject);
+            droppedObjectTarget.AddRange(droppedObjects);
 
-            drop(caughtObject, animation);
+            foreach (var droppedObject in droppedObjects)
+                applyDropAnimation(droppedObject, animation);
         }
 
-        private void drop(DrawablePalpableCatchHitObject d, DroppedObjectAnimation animation)
+        private void removeFromPlate(CaughtObject caughtObject, DroppedObjectAnimation animation)
         {
-            var originalX = d.X * Scale.X;
-            var startTime = Clock.CurrentTime;
+            var droppedObject = getDroppedObject(caughtObject);
 
-            d.Anchor = Anchor.TopLeft;
-            d.Position = caughtFruitContainer.ToSpaceOfOtherDrawable(d.DrawPosition, droppedObjectTarget);
+            if (!caughtFruitContainer.Remove(caughtObject))
+                throw new InvalidOperationException("Can only drop a caught object on the plate");
```

Comment:
Not sure if I'm understanding the intent right, would maybe reword to
```suggestion
                throw new InvalidOperationException("Can only drop objects that were previously caught on the plate");
```

Reply:
Actually this check is not needed anymore because this method was public but it is now private and the exception shouldn't occur.